src/controllers/authController.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from 'express';
+import * as jwt from 'jsonwebtoken';
+import { User } from '../models/user';
+import { getUserByLogin } from './userController';
+import { checkIfUnencryptedPasswordIsValid } from '../helpers/password.helper';
+
+export const login = async (req: Request, res: Response) => {
+
+  //Check if username and password are set
+  const { login, password } = req.body;
+
+  if (!(login && password)) {
+    res.sendStatus(400);
+  }
+  let user: User | null;
```

Comment:
Pourquoi le user peut être de type null ?

Reply:
Cause getUserByLogin could return "nothing" if the login doesn't match

Reply:
There is the problem mentioned by @ThomasF34  about the export at the end of the file instead of the declaration of const...